### PR TITLE
Added a README for the ml-agents package

### PR DESCRIPTION
**Pull request by vincentpierre**

### CODE REVIEW

File: ml-agents/README.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+pip install mlagents
+```
+
+or by running the following from the `ml-agents` directory of the repository:
+
+```
+pip install .
+```
+
+## Low level API
```
Comment (awjuliani) [inline ml-agents/README.md]: If we call this the "low level API" should we refer to the training api as the high level? Maybe just be explicit that this is `mlagents.envs` and the other is `mlagents.trainers`.

File: ml-agents/README.md [hunk 2]
```diff
@@ -101,7 +117,7 @@ variable named `env` in this example, can be used in the following way:
     a dictionary of strings to floats where the keys are the names of the
     `resetParameters` and the values are their corresponding float values.
     Define the reset parameters on the [Academy
-    Inspector](Learning-Environment-Design-Academy.md#academy-properties) window
+    Inspector](../docs/Learning-Environment-Design-Academy.md#academy-properties) window
```
Comment (dericp) [inline ml-agents/README.md]: Link won't work on PyPi.

File: ml-agents/README.md [hunk 3]
```diff
@@ -24,19 +41,18 @@ agent must either use an **External** brain or use a brain that is broadcasting
 actions for agents with external brains, but can only observe broadcasting
 brains (the information you receive for an agent is the same in both cases). See
 [Using the Broadcast
-Feature](Learning-Environment-Design-Brains.md#using-the-broadcast-feature).
+Feature](../docs/Learning-Environment-Design-Brains.md#using-the-broadcast-feature).
```
Comment (dericp) [inline ml-agents/README.md]: The problem with this relative path is that on the PyPi website itself, the `docs` directory won't be there. Maybe we should put a link to the file on GitHub?

File: ml-agents/README.md [hunk 4]
```diff
@@ -24,19 +41,18 @@ agent must either use an **External** brain or use a brain that is broadcasting
...
 
 For a simple example of using the Python API to interact with a Unity
-environment, see the Basic [Jupyter](Background-Jupyter.md) notebook
+environment, see the Basic [Jupyter](../docs/Background-Jupyter.md) notebook
```
Comment (dericp) [inline ml-agents/README.md]: Link won't work on PyPi.

File: ml-agents/README.md [hunk 5]
```diff
@@ -1,5 +1,22 @@
-# Python API
+# Unity ml-agents interface and trainers
 
+The `mlagents` package contains two components : The low level API which allows you to interact directly with a Unity Environment and a training component whcih allows you to train agents in Unity Environments using our implementations of reinforcement learning or imitation learning.
```
Comment (dericp) [inline ml-agents/README.md]: Line length should be capped at 80 chars long.

File: ml-agents/README.md [hunk 6]
```diff
@@ -1,5 +1,22 @@
...
+
+## Installation
+
+The `ml-agents` package can be installed using:
+
+```
```
Comment (dericp) [inline ml-agents/README.md]: Should include syntax highlighting instructions, so something like ```sh

File: ml-agents/README.md [hunk 7]
```diff
@@ -1,5 +1,22 @@
...
+pip install mlagents
+```
+
+or by running the following from the `ml-agents` directory of the repository:
+
+```
```
Comment (dericp) [inline ml-agents/README.md]: Should include syntax highlighting instructions, so something like ```sh

File: ml-agents/README.md [hunk 8]
```diff
@@ -1,5 +1,22 @@
...
+pip install .
+```
+
+## `mlagents.envs`
```
Comment (dericp) [inline ml-agents/README.md]: Markdown likes it better when there is a single blank line above and below a heading. I think there should be a blank line below this heading.

File: ml-agents/README.md [hunk 9]
```diff
@@ -22,21 +39,13 @@ To communicate with an agent in a Unity environment from a Python program, the
...
-simulation steps taking random actions, and closes the environment.
+brains (the information you receive for an agent is the same in both cases).
 
 _Notice: Currently communication between Unity and Python takes place over an
 open socket without authentication. As such, please make sure that the network
 where training takes place is secure. This will be addressed in a future
 release._
-
-## Loading a Unity Environment
+### Loading a Unity Environment
```
Comment (dericp) [inline ml-agents/README.md]: Add a blank line above this heading.

File: ml-agents/README.md [hunk 10]
```diff
@@ -116,6 +123,7 @@ variable named `env` in this example, can be used in the following way:
 
     For example, to access the BrainInfo belonging to a brain called
     'brain_name', and the BrainInfo field 'vector_observations':
+
```
Comment (dericp) [inline ml-agents/README.md]: Remove blank spaces on this line.

File: ml-agents/README.md [hunk 11]
```diff
@@ -127,6 +135,7 @@ variable named `env` in this example, can be used in the following way:
     and `value`. For example: If you have two external brains named `brain1` and
     `brain2` each with one agent taking two continuous actions, then you can
     have:
+
```
Comment (dericp) [inline ml-agents/README.md]: Remove blank spaces on this line.

File: ml-agents/README.md [hunk 12]
```diff
@@ -135,3 +144,30 @@ variable named `env` in this example, can be used in the following way:
 - **Close : `env.close()`**
   Sends a shutdown signal to the environment and closes the communication
   socket.
+
```
Comment (dericp) [inline ml-agents/README.md]: Markdown linter prefers only a single blank line between content and headers.

File: ml-agents/README.md [hunk 13]
```diff
@@ -135,3 +144,30 @@ variable named `env` in this example, can be used in the following way:
...
+
+
+## `mlagents.trainers`
+
+1. Open a command or terminal window.
+2. Run
```
Comment (dericp) [inline ml-agents/README.md]: Blank trailing space on this line.

File: ml-agents/README.md [hunk 14]
```diff
@@ -135,3 +144,30 @@ variable named `env` in this example, can be used in the following way:
...
   socket.
+
+
+
+## `mlagents.trainers`
+
+1. Open a command or terminal window.
+2. Run 
+
+```
```
Comment (dericp) [inline ml-agents/README.md]: Specify language as "sh".

File: ml-agents/README.md [hunk 15]
```diff
@@ -135,3 +144,30 @@ variable named `env` in this example, can be used in the following way:
...
+    - `<trainer-config-path>` is the relative or absolute filepath of the
+      trainer configuration. The defaults used by environments in the ML-Agents
+      SDK can be found in `config/trainer_config.yaml`.
+    - `<run-identifier>` is a string used to separate the results of different
+      training runs
+    - The `--train` flag tells `mlagents-learn` to run a training session (rather
+      than inference)
+    - `<environment-name>` __(Optional)__ is the path to the Unity executable you want		to train. __Note:__ If this argument is not passed, the training will be 
+       made through the editor.
+
```
Comment (dericp) [inline ml-agents/README.md]: Blank spaces on this line.

File: ml-agents/README.md [hunk 16]
```diff
@@ -135,3 +144,30 @@ variable named `env` in this example, can be used in the following way:
...
+    - `<run-identifier>` is a string used to separate the results of different
+      training runs
+    - The `--train` flag tells `mlagents-learn` to run a training session (rather
+      than inference)
+    - `<environment-name>` __(Optional)__ is the path to the Unity executable you want		to train. __Note:__ If this argument is not passed, the training will be 
+       made through the editor.
+       
+For more detailled documentation, check out the [ML-Agents toolkit documentation.](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Readme.md)
+
+
```
Comment (dericp) [inline ml-agents/README.md]: Unnecessary trailing new lines.

File: ml-agents/README.md [hunk 17]
```diff
@@ -135,3 +144,30 @@ variable named `env` in this example, can be used in the following way:
...
+   Where:
+    - `<trainer-config-path>` is the relative or absolute filepath of the
+      trainer configuration. The defaults used by environments in the ML-Agents
+      SDK can be found in `config/trainer_config.yaml`.
+    - `<run-identifier>` is a string used to separate the results of different
+      training runs
+    - The `--train` flag tells `mlagents-learn` to run a training session (rather
+      than inference)
+    - `<environment-name>` __(Optional)__ is the path to the Unity executable you want		to train. __Note:__ If this argument is not passed, the training will be
```
Comment (dericp) [inline ml-agents/README.md]: It looks like there's a hard tab in the middle of this line. Was that intentional?

Comment (vincentpierre) [reply ml-agents/README.md]: It was not

**Pull request merged**